ceph-volume: add --all flag to simple activate

jan--f opened the pull request:
This is intended to behave similarly to the lvm activate --all argument.
When passed, c-v will scan /etc/ceph/osd/ (or the location specified by
CEPH_VOLUME_SIMPLE_JSON_DIR) for json files (glob *.json) and call activate
for each file.  This should greatly ease the take-over of ceph-disk OSDs
with manual commands and deployment tools like DeepSea and ceph-ansible.

Signed-off-by: Jan Fajerski <jfajerski@suse.com>

jan--f commented:
@andrewschoen Happy to add some unit tests if you think this is useful.

jschmid1 commented:
Neat! I think this is useful for automated c-d -> c-v conversions.

andrewschoen commented:
@jan--f Yes, please add some unit tests to this. You could add a functional test for the ``--all`` flag here I believe as well. https://github.com/ceph/ceph/blob/master/src/ceph-volume/ceph_volume/tests/functional/simple/centos7/filestore/activate/test.yml#L26

The difficulty is that we can not test ``simple`` on the master branch so we'll have to wait until the backport to run those tests.

jan--f commented:
@andrewschoen I added a simple unit test and added functional tests specific for the `--all` functionality. Wasn't sure how to add them to the existing activate func-tests. Let me know if and how you prefer the latter.

andrewschoen commented:
@jan--f I don't think we need an entirely separate testing scenario for this. If you can edit one of the existing ``simple`` tests to use ``--all`` I think that'd be fine. You could do that here: https://github.com/ceph/ceph/blob/master/src/ceph-volume/ceph_volume/tests/functional/simple/centos7/filestore/activate/test.yml#L26

andrewschoen commented:
jenkins test ceph-volume tox

jan--f commented:
@andrewschoen like so?

jan--f commented:
@andrewschoen ping

andrewschoen commented:
jenkins test ceph-volume tox
